dapps/marketplace/src/pages/growth/WithEnrolmentModal.js

```diff
@@ -94,15 +102,19 @@ function withEnrolmentModal(WrappedComponent) {
       this.props.history.push(href)
     }
 
-    handleClick(e, enrollmentStatus, walletPresent) {
+    handleClick(e, enrollmentStatus) {
       e.preventDefault()
 
+      const { isLoggedIn } = this.props
+
       if (mobileDevice() !== null && !rewardsOnMobileEnabled) {
         this.setState({
           open: true,
           stage: 'NotSupportedOnMobile'
         })
-      } else if (!walletPresent) {
+      } else if (!isLoggedIn) {
+        const { pathname, search } = this.props.location
+        sessionStore.set('getStartedRedirect', { pathname, search })
```

Comment:
Might want to check if this is already set.  I think this is being utilized by mobile to send users to `/campaigns`.